api/v1alpha1/menu_types.go

```diff
@@ -82,9 +82,6 @@ type MenuSpec struct {
 	// 父菜单 ID
 	// +optional
 	Parent string `json:"parent,omitempty"`
-	//对应 k8s 资源的 name
-	// +optional
-	ParentOwnerReferences metav1.OwnerReference `json:"parentOwnerReferences,omitempty"`
```

Comment:
@Carrotzpc any idea?

kubernetes provide default OwnerReference in `metadata.ownerReferences`.
The default rule is that:
1. If A's `metadata.ownerReferences` is B, then when B is deleted, A will be deleted automatically.
2. A and B must be in the same namespace.

I don't know if this meets our front-end design requirements.

Reply:
As discussed, this can be deleted as well. @mengqn021

Reply:
Previously, we were using the metadata.ownerReferences field, where the uid in metadata.ownerReferences was mandatory. This means that we had to create the parent menu first and obtain the uid of the parent menu before creating the child menu. This made the deployment process more cumbersome. In order to simplify the deployment, we added a new field in the spec called ParentOwnerReferences, which does not require the uid.

Reply:
When parent menu got deleted, the children menus must be deleted as well. In kubernetes, the standard way is to use `OwnerReference`  which we should follow.  And as i said, `creating menus` is a big stuff. If this is done without the user's manual approval, the overall service menu provided by kubebb can be in chaos.

So I think we should :

1. Keep `OwnerReference` in metadata 
   - remove the `ParentOwnerReference`
   -  remove `ID``Parent` which seems useless when rendering menus in bff server???  @Carrotzpc 
2. The component published by low code server can keep a `menu.json` under its helm charts.Take [bc-apis](https://github.com/kubebb/components/tree/main/charts/bc-apis) as an example, keep a json  at`bc-apis/menus.json` 

3. Add a new filed `AutoMenu bool` under `ComponentPlan.Spec`. When set to true, trigger a reconcile logic to process the `menus.json` and create menus CRs accordingly. ( @mengqn961127  How about this? We create menus with a `Approval` from admin)

Reply:
As discussed, we 
- remove `ParentOwnerReference` but keep `ID` and `Parent`. Process menu.json in kubebb/core
- add a new field in `AutoMenu`